File: src/main/kotlin/com/indefiniteStudies/model/entities/Task.kt
package com.indefiniteStudies.model.entities

import org.jetbrains.exposed.sql.Column
import org.jetbrains.exposed.sql.Table

object Task : Table("tasks") {
    val id: Column<Int> = integer("id").autoIncrement()
    val name: Column<String> = varchar("name", 100)
    val state: Column<String> = varchar("state", 100)
}

File: internal/db/entities/task.go
package entities

type Task struct {
	Id    int
	Name  string
	State string
}

const (
	TASK_STATE_NEW     string = "NEW"
	TASK_STATE_DONE    string = "DONE"
	TASK_STATE_DELETED string = "DELETED"
)

func GetPossibleTaskStates() []string {
	return []string{TASK_STATE_NEW, TASK_STATE_DONE, TASK_STATE_DELETED}
}
